### PR TITLE
Initial Peer and Artwork class code

### PR DESCRIPTION
**Pull request by thompalex**
Our Peer class will have all of the functions that manage connections to other nodes. Currently, it has some baseline code for:
- commissioning an art piece
- sending a notification that the deadline has passed

Note that these are not complete functions, since we still need implementations of the join_ipfs_network and ipfs_publish functions (if the libraries we decide to use have these sorts of functions)

We can also pivot to send these messages separately, but it does seem like the libraries we are thinking about using will handle a good amount of the organization for messaging between peers.

### CODE REVIEW

**Comment by nghi01** [comment]
I think it's a good start and we should aim to make this part of the code testable in our project.

File: src/main/py/commission/artwork.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+class Artwork:
+    """
+    Class to manage artwork commissions
+    """
+
+    def __init__(self, width, height, wait_time):
+        """
+        Initializes an instance of the Artwork class.
+        - width (float): The width of the artwork.
```
Comment (zimmermatt) [inline src/main/py/commission/artwork.py:17]: It's always helpful to know what the unit is (e.g., cm, inches, pixels). We probably  want pixels in this case.

As far as specifying the type, I would just use the Python type hint feature in the signature, even though we're not running a type checker at this point. The other option, given we're not type checking would be to just not specify anything at this point.

File: src/main/py/commission/artwork.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        - height (float): The height of the artwork.
+        - wait_time (float): The wait time for the artwork.
```
Comment (zimmermatt) [inline src/main/py/commission/artwork.py:19]: I would use the `datetime` class.

File: src/main/py/commission/artwork.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        Returns the file descriptor for the artwork.
+        """
+        return self.ipfs_file_descriptor
+
+    def set_complete(self):
+        """
+        Sets the commission status to complete.
+        """
+        self.commission_complete = True
+        return self
```
Comment (zimmermatt) [inline src/main/py/commission/artwork.py:50]: Probably don't need to return anything for this method.

File: src/main/py/peer/peer.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    """
+
+    logger = logging.getLogger("Peer")
+
+    def __init__(self, network_address: str, port: int, ipfs) -> None:
+        """
+        Initialize the Peer class by joining the IPFS network.
+
+        Params:
+        - network_address (str): The network address to join.
```
Comment (zimmermatt) [inline src/main/py/peer/peer.py:27]: I would use the `ipaddress` class.

File: src/test/py/peer/peer_test.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        "threading.Timer",
+        side_effect=lambda delay, func, args: Timer(0.001, func, args),
+    )
+    def test_commission_art_piece(self, mock_input, mock_timer):
+        """
+        Test case for the commission_art_piece method of the Peer class.
+        This test verifies that the commission_art_piece method correctly adds a commission,
+        publishes it on IPFS, and schedules and sends a deadline notice.
+        """
+        print(mock_input, mock_timer)
```
Comment (zimmermatt) [inline src/test/py/peer/peer_test.py:56]: Should avoid printing, except for temporary quick debug.

File: src/test/py/peer/peer_test.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        self.peer.commission_art_piece()
+        self.assertEqual(len(self.peer.commissions), 1)
+        commission = self.peer.commissions[0]
+        self.assertEqual(commission.width, 10)
+        self.assertEqual(commission.height, 20)
+        self.assertEqual(commission.wait_time, 10)
+        self.mock_ipfs.ipfs_publish.assert_called_once_with(
+            "placeholder_ipfs_file_descriptor", commission
+        )
+        time.sleep(0.02)
```
Comment (zimmermatt) [inline src/test/py/peer/peer_test.py:66]: This is okay for now, but relying on `sleep` for async code yields fragile tests typically. As this evolves we can look at ways of structuring the code and/or using more predictable concurrency testing techniques.

**Review by zimmermatt — APPROVED** [review]
Overall great initial effort. I skipped thorough review of sections that will likely change given that we now have a specific peer-to-peer framework chosen.

I'm going to go ahead and merge this. The comments I made here can be addressed in upcoming PRs.

**Pull request merged**